packages/api/tests/endpoints/token.js

```diff
@@ -244,8 +229,117 @@ test.serial('When using Earthdata Login, GET /token with a code updates the user
 
   await handleRequest(request, mockOAuth2Provider);
 
-  const userAfter = await userModel.get({ userName });
+  const tokenAfter = await accessTokenModel.get({ accessToken });
+
+  t.is(tokenAfter.accessToken, accessToken);
+  t.is(tokenAfter.refreshToken, refreshToken);
+});
+
+test.serial('GET /refresh without a token results in an authorization failure response', async (t) => {
+  const request = {
+    httpMethod: 'POST',
+    resource: '/refresh'
+  };
+
+  const response = await handleRequest(request);
+
+  t.is(response.statusCode, 400);
+  t.is(JSON.parse(response.body).message, 'Request requires a token');
+});
+
+test.serial('GET /refresh with an invalid token results in an authorization failure response', async (t) => {
+  const request = {
+    httpMethod: 'POST',
+    resource: '/refresh',
+    body: JSON.stringify({
+      token: 'InvalidToken'
+    })
+  };
+
+  const response = await handleRequest(request);
+
+  assertions.isInvalidAccessTokenResponse(t, response);
+});
+
+test.serial('GET /refresh with an non-existent token results in an authorization failure response', async (t) => {
+  const accessTokenRecord = fakeAccessTokenFactory();
+  const jwtToken = createJwtToken(accessTokenRecord);
+
+  const request = {
+    httpMethod: 'POST',
+    resource: '/refresh',
+    body: JSON.stringify({
+      token: jwtToken
+    })
+  };
+
+  const response = await handleRequest(request);
+
+  assertions.isInvalidAccessTokenResponse(t, response);
+});
+
+test.serial('GET /refresh returns 500 if refresh token request fails', async (t) => {
+  const mockOAuth2Provider = {
+    refreshAccessToken: async () => {
+      throw new Error('Refresh token request failed');
+    }
+  };
+
+  const userRecord = fakeUserFactory();
+  await userModel.create(userRecord);
+
+  const initialTokenRecord = fakeAccessTokenFactory({ username: userRecord.userName });
+  await accessTokenModel.create(initialTokenRecord);
+
+  const requestJwtToken = createJwtToken(initialTokenRecord);
+
+  const request = {
+    httpMethod: 'POST',
+    resource: '/refresh',
+    body: JSON.stringify({
+      token: requestJwtToken
+    })
+  };
+
+  const response = await handleRequest(request, mockOAuth2Provider);
+  t.is(response.statusCode, 500);
+});
+
+test.serial('GET /refresh with a valid token returns a refreshed token', async (t) => {
+  const refreshedTokenRecord = fakeAccessTokenFactory();
+  const refreshedJwtToken = createJwtToken(refreshedTokenRecord);
+
+  const mockOAuth2Provider = {
+    refreshAccessToken: async () => refreshedTokenRecord
+  };
+
+  const userRecord = fakeUserFactory();
+  await userModel.create(userRecord);
 
-  t.is(userAfter.refresh, 'my-refresh-token');
-  t.is(userAfter.password, 'my-access-token');
+  const initialTokenRecord = fakeAccessTokenFactory({ username: userRecord.userName });
```

Comment:
The answer here is the same as https://github.com/nasa/cumulus/pull/717#discussion_r237532754. But I will clean up the code here to avoid the confusion